### PR TITLE
test: Add E2E tests for seated attendee rescheduling or cancelling bookings

### LINKED ISSUE #9362: [CAL-1876] Make E2E test: Attendee #1 Should be able to cancel his booking
<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-1876](https://linear.app/calcom/issue/CAL-1876/make-e2e-test-attendee-1-should-be-able-to-cancel-his-booking)</sub>

### PR DESCRIPTION
**Pull request by rjackson**
## What does this PR do?

This PR adds or expands end-to-end tests to cover cancellation or reschedule actions taken by the attendees or seated events:

- booking-seats.e2e.ts
  - Attendees can cancel a seated event time slot
    - Attendee no. 1 should be able to cancel their booking
    - All attendees cancelling should delete the booking for the user
  - Should reschedule booking with seats (updated to make sure the new booking is set up correctly)
- reschedule.e2e.ts
  - Attendee should be able to reschedule a booking
- booking-pages.e2e.ts
  - Can cancel the recently created booking and rebook the same timeslot (updated to remove dependence on localised text)
- dynamic-booking-pages.e2e.ts
  - Can cancel the recently created booking (updated to remove dependence on localised text)

There are a few other small changes and fixes alongside this PR:

- Updated test utils and new booking UI to avoid getting stuck on disabled time slots
    This was a problem whilst I was exploring non-seated event tests, as I had misunderstood the scope of these issues. Whilst it caused no breakages at present, the test utils should be able to skip disabled time slots in the new booking UI
- Add explicit test IDs for cancel inputs
    The same `testid` was in-use by two cancel buttons: One which opens the form with "Cancellation reason", and the other that actually submits the cancel. For clarity and to ease future test writing, I've given these separate IDs. I also added a `testid` for the "Cancellation reason" field
- Fixed backward-compatibility logic for `bookingFields` to avoid loading attendees as "guests" for event types with guests disabled (which is the case for seated bookings)
    This caused a lot of trouble when I was expanding the "reschedule" test to ensure the rescheduled attendee's booking was created correctly. According to my expanded test it was not, but when manually testing it was – with this backward-compatibility logic being responsible for that difference.
- Updated some existing tests to remove their dependence on localised text

(Following section struck out as it relates to unused functionality)

<strike>
And there remains one major issue, left unaddressed by this PR: Cancelling a booking from the `/booking/${seatRef}` endpoint doesn't work correctly, it:

1. Cancels the whole booking for all attendees (very unpleasant!)
2.~Shows a 404 page to the user instead of the "You are no longer attending this event" messaging (#8244)

I have written the cancellation-related tests to cover both the `/booking/${bookingRef}?seatReferenceUid=${seatRef}` URL and the seatRef URL, but left the seatRef tests being skipped default.

The seatRef tests can be enabled by running the tests with `E2E_TEST_SEATED_CANCELLATIONS=1`, for example to run _just_ those tests:

```txt
E2E_TEST_SEATED_CANCELLATIONS=1 yarn e2e booking-seats.e2e.ts -g "Attendees can cancel a seated event time slot"
```
</strike>

Fixes #9362 #9363 #9364
/claim #9362 
/claim #9363 
/claim #9364

## Type of change

- Bug fix (non-breaking change which fixes an issue)
- Chore (refactoring code, technical debt, workflow improvements)

## How should this be tested?

- [x] Run automated e2e tests
  - I did have some failures locally, but these were timeouts in unrelated and unchanged files. Hopefully all will run smoothly in CI

## Mandatory Tasks
- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

## Checklist

(all complete)

### CODE REVIEW

File: apps/web/playwright/lib/testUtils.ts [hunk 1]
```diff
@@ -92,7 +92,7 @@ export async function selectFirstAvailableTimeSlotNextMonth(page: Page) {
   await page.waitForTimeout(1000);
   // TODO: Find out why the first day is always booked on tests
   await page.locator('[data-testid="day"][data-disabled="false"]').nth(1).click();
-  await page.locator('[data-testid="time"]').nth(0).click();
+  await page.locator('[data-testid="time"][data-disabled="false"]').nth(0).click();
```
Comment (roae) [inline apps/web/playwright/lib/testUtils.ts:95]: nice catch!

File: apps/web/playwright/booking-seats.e2e.ts [hunk 2]
```diff
@@ -105,25 +113,108 @@ testBothBookers.describe("Booking with Seats", (bookerVariant) => {
...
+
+        await page.locator('[data-testid="cancel"]').click();
+        await page.fill('[data-testid="cancel_reason"]', "Double booked!");
+        await page.locator('[data-testid="confirm_cancel"]').click();
+        await page.waitForLoadState("networkidle");
+
+        await expect(page).toHaveURL(/.*booking/);
+
+        const cancelledHeadline = await page.locator('[data-testid="cancelled-headline"]').innerText();
+        expect(cancelledHeadline).toBe("You are no longer attending this event");
```
Comment (roae) [inline apps/web/playwright/booking-seats.e2e.ts:166]: @zomars Can we rely on this message? or is it advisable to have another validation method? 

If this message changes the test will fail

Comment (zomars) [reply apps/web/playwright/booking-seats.e2e.ts:166]: Agreed. We should expect the container to show. Something like `data-testid="error-message"`

Comment (rjackson) [reply apps/web/playwright/booking-seats.e2e.ts:166]: Good shout!

I ended up copying this `cancelledHeadline` pattern from existing tests:

https://github.com/calcom/cal.com/blob/974036239f204a21cc974026aa7366111b9012ca/apps/web/playwright/booking-pages.e2e.ts#L118-L120

Are we happy updating those existing usages alongside this PR, so we have consistency across tests?

Comment (roae) [reply apps/web/playwright/booking-seats.e2e.ts:166]: yes, just add it to the description for reference as you did with the others please.

Comment (rjackson) [reply apps/web/playwright/booking-seats.e2e.ts]: Done in 09b29ac03681d468db160233cedc296095afbb4c

**Comment by roae** [comment]
@rjackson I'm trying to replicate the issue canceling the booking with the endpoint `/booking/${seatRef}` but, I can find where that endpoint is used.

**Comment by rjackson** [comment]
> @rjackson I'm trying to replicate the issue canceling the booking with the endpoint `/booking/${seatRef}` but, I can find where that endpoint is used.

Hmm, I thought I got that link from the emails that go out – but looking at it again, I can see that emails use `/reschedule/${seatUid}` for reschedules, but `/booking/${bookingUid}?seatReferenceUid=${seatUid}` for cancellations:

https://github.com/calcom/cal.com/blob/09b29ac03681d468db160233cedc296095afbb4c/packages/lib/CalEventParser.ts#L177

I can't find anywhere else that might cause it, unless it somehow ever ends up as `calEvent.uid`? I can't spot that happening either.

Maybe the reschedule usage is where I originally saw this type of URL structure and then later noticed that `/booking/` also supports it when accessed directly in the browser? Apologies for misleading us if that's the case!

I'd say that lowers the priority of the "bug", but I would point out that `/booking/${seatUid}` does appear to be supported by the codebase so it would be nice if we can make it work end-to-end, or we redirect that back to the canonical booking URL? (I have a separate issue that I've reported privately that _might_ warrant making `/booking/${seatUid}` the primary link for seated participants, but that's beyond the scope of this issue).

Whatever we decide, as this URL appears to be an unexposed "feature"/"bug" I don't think we ought to have it represented in these tests. That can change if we ever do decide to formally support it, of course.

**Review by roae — APPROVED** [review]
Great job @rjackson, thanks for the contribution! 🙏

**Pull request merged**